network/p2p/p2p_pubsub.go

```diff
@@ -157,62 +143,89 @@ func (n *p2pNetwork) Subscribe(pk spectypes.ValidatorPK) error {
 	if !n.isReady() {
 		return p2pprotocol.ErrNetworkIsNotReady
 	}
+
+	share := n.validatorStore.Validator(pk[:])
+	if share == nil {
+		return fmt.Errorf("could not find share for validator %s", hex.EncodeToString(pk[:]))
+	}
+
+	err := n.subscribeCommittee(share.CommitteeID())
+	if err != nil {
+		return fmt.Errorf("could not subscribe to committee: %w", err)
+	}
+	if !n.cfg.Network.PastAlanFork() {
+		return n.subscribeValidator(pk)
+	}
```

Comment:
Just to confirm, we are subscribing to the committee-related subnet even though we may not already be on post-fork, right? Is this for safety?

Reply:
Since not subscribing to committee subnets and then subscribing right at the fork epoch might result in us not having enough peers at the fork epoch, we subscribe in advance to both.

Reply:
Got it. Great! Thanks for the clarification